cmd/prometheus/main.go

```diff
@@ -971,6 +973,11 @@ func main() {
 		}, {
 			name: "scrape_sd",
 			reloader: func(cfg *config.Config) error {
+				if agentMode {
```

Comment:
Instead of multiple if-statements, can you pull out the non-agent reloaders into a second set which are `append`-ed to the list as necessary?